src/utils/useStateReadState/index.ts

```diff
@@ -1,123 +1,45 @@
 import { useEffect, useState } from 'react'
-import { QueryHookOptions, useApolloClient } from '@apollo/client'
-import type { CID } from '@glif/filecoin-wallet-provider'
+import { LotusActorState } from '@glif/filecoin-actor-utils'
+import { validateAddressString } from '@glif/filecoin-address'
 
-import { Address, AddressDocument, AddressQuery } from '../../generated/graphql'
-import { decodeActorCID } from '..'
 import { useEnvironment } from '../../services/EnvironmentProvider'
 
-export type LotusRPCActorState<T> = {
-  Balance: string
-  Code: CID
-  State: T
+interface UseStateReadStateResult<T> {
+  data: LotusActorState<T> | null
+  loading: boolean
+  notFound: boolean
+  error: Error | null
 }
 
-export type MsigState<T = Address> = {
-  InitialBalance: string
-  AvailableBalance: string
-  NextTxnID: number
-  NumApprovalsThreshold: number
-  Signers: T[]
-  StartEpoch: number
-  UnlockDuration: number
-}
-
-const emptyActorState: LotusRPCActorState<any> = {
-  Balance: '0',
-  Code: { '/': '' },
-  State: null
-}
-
-/////// implemented until https://github.com/glifio/graph/issues/33
-export const useStateReadStateQuery = <T = any>(
-  baseOptions: QueryHookOptions<
-    T,
-    {
-      address: string
-    }
-  >
```

Comment:
The name and params made it seem like this was an apollo query, which it's not. The params like `skip` that could be passed were not handled either, only the address was grabbed from the `baseOptions`.